core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -1019,6 +1030,21 @@ private NativeImageSourceJarBuildItem buildNativeImageThinJar(CurateOutcomeBuild
             removedArtifacts.add("org.graalvm.sdk:word");
             removedArtifacts.add("org.graalvm.sdk:collections");
 
+            nativeImageRunnerBuildItem.getBuildRunner()
+                    .setup(processInheritIODisabled.isPresent() || processInheritIODisabledBuildItem.isPresent());
```

Comment:
I think we tried hard to avoid calling this over and over? (ping @Karm as he was the one who fought this fight)

Is it possible to somehow get this information once and for all and propagate it to where it's needed?